modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/AndesAckData.java

```diff
@@ -18,17 +18,33 @@
 
 package org.wso2.andes.kernel;
 
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.wso2.andes.kernel.subscription.AndesSubscription;
+import org.wso2.andes.metrics.MetricsConstants;
+import org.wso2.andes.tools.utils.MessageTracer;
+import org.wso2.carbon.metrics.manager.Counter;
+import org.wso2.carbon.metrics.manager.Level;
+import org.wso2.carbon.metrics.manager.Meter;
+import org.wso2.carbon.metrics.manager.MetricManager;
+
 import java.util.UUID;
 
 /**
  * Wrapper class of message acknowledgment data publish to disruptor
  */
 public class AndesAckData {
 
+    private static Log log = LogFactory.getLog(AndesAckData.class);
     /**
-     * Acknowledged message
+     * Id of acknowledged message
```

Comment:
Yes. This is because when we create AckData, what we set is messageId (from transport we get to which message ACK came). We need to set the reference when disruptor thread executes only (inside a disruptor handler). Thus we need both.